src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -105,11 +106,8 @@ double HMM<Distribution>::Train(const std::vector<arma::mat>& dataSeq)
   for (size_t seq = 0; seq < dataSeq.size(); seq++)
   {
     totalLength += dataSeq[seq].n_cols;
-
-    if (dataSeq[seq].n_rows != dimensionality)
-      Log::Fatal << "HMM::Train(): data sequence " << seq << " has "
-          << "dimensionality " << dataSeq[seq].n_rows << " (expected "
-          << dimensionality << " dimensions)." << std::endl;
+      
+      util::CheckSameDimensionality(dataSeq[seq], dimensionality, "HMM::Train()");
```

Comment:
The problem I see with this is that the user has no idea, what went wrong, all he get's is `HMM::Train()`, so we should add the same message as we had before.